doc/architecture/decisions/0002-pathogens-portal-2-0-architecture.md

```diff
@@ -0,0 +1,150 @@
+| **Database** | PostgreSQL | Reliable, performant, and supports advanced features such as full-text search. Integrates seamlessly with Django ORM.                                            |
+
+### Alternatives Considered
+
+| Option                                               | Reason for Rejection                                                                                                                       |
+| ---------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| **Monolithic Django app** (Django + templates) | Creates tighter coupling between frontend and backend parts, not made to scale components independently and more difficult to implement a complete frontend frameworks. |
+| **Headless CMS** (Strapi, Directus)                  | Adds operational overhead and limits customisation in terms of workflows, data validation, and complex integrations.                       |
+| **Continue with Hugo + JS + APIs**                   | Already showing scalability and maintainability limits. Poor support for dynamic workflows, user roles, and content contribution features. |
+| **Node.js (Express)** for backend                    | More boilerplate, lacks built-in features like Django admin. Team familiarity is lower.                                                    |
+
+
+This decision positions Pathogens Portal 2.0 for long-term maintainability and expansion, with the ability to support richer content, data interactivity, editorial workflows, and integration with SciLifeLab’s broader data infrastructure.
+
+
+## Consequences
+
+Adopting a service-oriented architecture (SOA) with decoupled frontend, backend, and database components introduces both **benefits** and **trade-offs**.
+
+### Positive Outcomes
+
+* **Improved modularity:** Each component (frontend, backend, database) can be developed, tested, and deployed independently.
+* **Scalable design:** The system can scale horizontally by distributing workload across services, enabling better handling of increased user traffic or data volumes.
+* **Future-ready:** The architecture provides a strong foundation for supporting new features such as:
+
+  * Automated content and data update workflows without requiring portal releases
+  * AI-powered content summarisation and intelligent data processing
+  * Real-time or scheduled data visualisations
+  * Editorial workflows with automated publishing pipelines
+  * Federated data integration via APIs
+  * End-to-end automation from raw data ingestion to visualisation generation
+
+* **Flexible Integrations**: External systems like Portal Storage Solution, IDS, and domain-specific repositories can be plugged in with minimal effort.
+
+
+### Risks & Challenges
```

Comment:
I'd add that this approach complicates testing and debugging of the whole service as you need to have knowledge in how to run each service individually.

Additionaly I would suggest to rely on api documentation as a communication point between frontend and backend teams

Reply:
This wasn't mentioned during our discussions but it is also true